pkg/bff/db/announcements_test.go

```diff
@@ -0,0 +1,150 @@
+package db_test
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"github.com/trisacrypto/directory/pkg/bff/api/v1"
+	. "github.com/trisacrypto/directory/pkg/bff/db"
+)
+
+func (s *dbTestSuite) TestAnnouncements() {
+	// Test creating and retrieving announcements
+	var err error
+	require := s.Require()
+
+	// Announcements should implement the Collection interface
+	require.Equal(NamespaceAnnouncements, s.db.Announcements().Namespace())
+
+	ctx, cancel := context.WithTimeout(context.Background(), 20*time.Second)
+	defer cancel()
+
+	// There should be nothing in the database at the start of the test.
+	recent, err := s.db.Announcements().Recent(ctx, 10000, time.Time{})
+	require.NoError(err, "could not fetch recent announcements")
+	require.Len(recent.Announcements, 0, "expected no announcements returned")
+	require.Empty(recent.LastUpdated, "expected last updated to be zero-valued")
+
+	// Load announcements fixtures
+	fixture, err := loadAnnouncements()
+	require.NoError(err, "could not load announcement fixtures")
+	require.Len(fixture, 10, "fixtures have changed without test update")
+
+	// Post each fixture one at a time and ensure that we can fetch them all
+	ids := make([]string, 0, 10)
+	for i, post := range fixture {
+		id, err := s.db.Announcements().Post(ctx, post)
+		require.NoError(err, "could not post announcement %d", i)
+		require.NotEmpty(id, "expected an ordered kid to be assigned")
+		ids = append(ids, id)
+
+		// Ensure there is at least 1 second between each post or the ksuid will
+		// identify the posts as concurrent and produce an incorrect ordering.
+		// Normally posts won't be published within the same second, but this is a bad
+		// situation for tests because it means that this test will take a min of 10
+		// seconds to run, which adds up in CI and beyond ...
+		time.Sleep(1 * time.Second)
```

Comment:
Le sigh

Reply:
That is annoying, the only way I can think of to avoid this is to not use the ksuids for ordering.

Reply:
I know right? So annoying - and it feels like something that should be configurable. I found lots of git issues complaining about it and related issues.